Custom lifetime error for `impl` item doesn't conform to `trait`

estebank opened the pull request:
Partly addresses #42706, #41343, fix #40900.

estebank committed:
Custom lifetime error for `impl` item doesn't conform to `trait`

rust-highfive commented:
r? @zackmdavis

(rust_highfive has picked a reviewer for you, use r? to override)

rust-highfive commented:
The job `x86_64-gnu-llvm-6.0` of your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=9866&view=logs&jobId=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/9866/logs/49), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/9866/logs/49)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-10-03T18:47:59.0592524Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-10-03T18:47:59.0762781Z ##[command]git config gc.auto 0
2019-10-03T18:47:59.0843078Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-10-03T18:47:59.0904023Z ##[command]git config --get-all http.proxy
2019-10-03T18:47:59.1057899Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/65068/merge:refs/remotes/pull/65068/merge
---
2019-10-03T19:51:35.8635323Z .................................................................................................... 1500/9098
2019-10-03T19:51:42.9262973Z .................................................................................................... 1600/9098
2019-10-03T19:51:52.6297043Z .................................................................................................... 1700/9098
2019-10-03T19:52:02.0553428Z .....i...............i.............................................................................. 1800/9098
2019-10-03T19:52:09.3663060Z ................................................................................................iiii 1900/9098
2019-10-03T19:52:26.5900159Z i................................................................................................... 2000/9098
2019-10-03T19:52:35.5033255Z .................................................................................................... 2200/9098
2019-10-03T19:52:38.2501429Z .................................................................................................... 2300/9098
2019-10-03T19:52:44.7862991Z .................................................................................................... 2400/9098
2019-10-03T19:52:50.5291687Z .................................................................................................... 2500/9098
---
2019-10-03T19:55:49.5788517Z ...................................................................................i...............i 4700/9098
2019-10-03T19:55:58.1479456Z .................................................................................................... 4800/9098
2019-10-03T19:56:08.8864401Z .................................................................................................... 4900/9098
2019-10-03T19:56:14.6748707Z .................................................................................................... 5000/9098
2019-10-03T19:56:27.0948227Z ...........................................................................ii.ii.................... 5100/9098
2019-10-03T19:56:36.9407340Z .................................................................................................... 5300/9098
2019-10-03T19:56:46.9567469Z .................................................................................................... 5400/9098
2019-10-03T19:56:54.1327424Z ..........................................i......................................................... 5500/9098
2019-10-03T19:57:01.0972801Z .................................................................................................... 5600/9098
2019-10-03T19:57:01.0972801Z .................................................................................................... 5600/9098
2019-10-03T19:57:12.1199612Z .................................................................................................... 5700/9098
2019-10-03T19:57:22.6740583Z .......................................ii...i..ii...........i....................................... 5800/9098
2019-10-03T19:57:46.1705835Z .................................................................................................... 6000/9098
2019-10-03T19:57:55.7843815Z .................................................................................................... 6100/9098
2019-10-03T19:57:55.7843815Z .................................................................................................... 6100/9098
2019-10-03T19:58:09.0334048Z ............................................i..ii................................................... 6200/9098
2019-10-03T19:58:31.4434179Z .................................................................................................... 6400/9098
2019-10-03T19:58:33.6730957Z ....i............................................................................................... 6500/9098
2019-10-03T19:58:35.9642522Z ............................................................................i....................... 6600/9098
2019-10-03T19:58:38.7490385Z .................................................................................................... 6700/9098
---
2019-10-03T20:02:53.7735281Z failures:
2019-10-03T20:02:53.7775321Z 
2019-10-03T20:02:53.7776459Z ---- [ui] ui/lifetimes/lifetime-mismatch-between-trait-and-impl.rs stdout ----
2019-10-03T20:02:53.7776533Z normalized stderr:
2019-10-03T20:02:53.7777111Z error: `impl` item doesn't match `trait` item
2019-10-03T20:02:53.7777684Z   --> $DIR/lifetime-mismatch-between-trait-and-impl.rs:6:5
2019-10-03T20:02:53.7777896Z    |
2019-10-03T20:02:53.7778312Z LL |     fn foo<'a>(x: &'a i32, y: &'a i32) -> &'a i32 {
2019-10-03T20:02:53.7778775Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ found fn(&i32, &i32) -> &i32
2019-10-03T20:02:53.7779007Z    |
2019-10-03T20:02:53.7779332Z    = note: expected: fn(&i32, &'a i32) -> &'a i32
2019-10-03T20:02:53.7779739Z               found: fn(&i32, &i32) -> &i32
2019-10-03T20:02:53.7780031Z error: aborting due to previous error
2019-10-03T20:02:53.7780062Z 
2019-10-03T20:02:53.7780090Z 
2019-10-03T20:02:53.7780117Z 
2019-10-03T20:02:53.7780117Z 
2019-10-03T20:02:53.7780160Z 
2019-10-03T20:02:53.7780209Z The actual stderr differed from the expected stderr.
2019-10-03T20:02:53.7780617Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lifetimes/lifetime-mismatch-between-trait-and-impl/lifetime-mismatch-between-trait-and-impl.stderr
2019-10-03T20:02:53.7780900Z To update references, rerun the tests and pass the `--bless` flag
2019-10-03T20:02:53.7781472Z To only update this specific test, also pass `--test-args lifetimes/lifetime-mismatch-between-trait-and-impl.rs`
2019-10-03T20:02:53.7782263Z error: 1 errors occurred comparing output.
2019-10-03T20:02:53.7782324Z status: exit code: 1
2019-10-03T20:02:53.7782324Z status: exit code: 1
2019-10-03T20:02:53.7783318Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/lifetimes/lifetime-mismatch-between-trait-and-impl.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lifetimes/lifetime-mismatch-between-trait-and-impl" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/lifetimes/lifetime-mismatch-between-trait-and-impl/auxiliary" "-A" "unused"
2019-10-03T20:02:53.7783700Z ------------------------------------------
2019-10-03T20:02:53.7783749Z 
2019-10-03T20:02:53.7783998Z ------------------------------------------
2019-10-03T20:02:53.7784067Z stderr:
2019-10-03T20:02:53.7784067Z stderr:
2019-10-03T20:02:53.7784640Z ------------------------------------------
2019-10-03T20:02:53.7784860Z error: `impl` item doesn't match `trait` item
2019-10-03T20:02:53.7785625Z   --> /checkout/src/test/ui/lifetimes/lifetime-mismatch-between-trait-and-impl.rs:6:5
2019-10-03T20:02:53.7785820Z    |
2019-10-03T20:02:53.7786140Z LL |     fn foo<'a>(x: &'a i32, y: &'a i32) -> &'a i32 {
2019-10-03T20:02:53.7786429Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ found fn(&i32, &i32) -> &i32
2019-10-03T20:02:53.7786622Z    |
2019-10-03T20:02:53.7786980Z    = note: expected: fn(&i32, &'a i32) -> &'a i32
2019-10-03T20:02:53.7787238Z               found: fn(&i32, &i32) -> &i32
2019-10-03T20:02:53.7787465Z error: aborting due to previous error
2019-10-03T20:02:53.7787527Z 
2019-10-03T20:02:53.7787578Z 
2019-10-03T20:02:53.7787865Z ------------------------------------------
---
2019-10-03T20:02:53.7821693Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:537:22
2019-10-03T20:02:53.7822027Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-10-03T20:02:53.7839706Z 
2019-10-03T20:02:53.7839803Z 
2019-10-03T20:02:53.7844440Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-10-03T20:02:53.7844930Z 
2019-10-03T20:02:53.7844966Z 
2019-10-03T20:02:54.4989159Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2019-10-03T20:02:54.4989541Z Build completed unsuccessfully in 1:07:26
2019-10-03T20:02:54.4989541Z Build completed unsuccessfully in 1:07:26
2019-10-03T20:02:54.4989688Z == clock drift check ==
2019-10-03T20:02:54.4989764Z   local time: Thu Oct  3 20:02:53 UTC 2019
2019-10-03T20:02:54.4989816Z   network time: Thu, 03 Oct 2019 20:02:53 GMT
2019-10-03T20:02:54.4989864Z == end clock drift check ==
2019-10-03T20:02:54.9551787Z ##[error]Bash exited with code '1'.
2019-10-03T20:02:54.9603562Z ##[section]Starting: Checkout
2019-10-03T20:02:54.9605887Z ==============================================================================
2019-10-03T20:02:54.9605962Z Task         : Get sources
2019-10-03T20:02:54.9606026Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

Centril commented on src/librustc/infer/error_reporting/nice_region_error/trait_impl_difference.rs:
```diff
@@ -0,0 +1,53 @@
+                            ValuePairs::Types(sup_expected_found),
+                        ) = (&sub_trace.values, &sup_trace.values) {
+                            if sup_expected_found == sub_expected_found {
+                                let sp = var_origin.span();
```
Consider moving this out to a free function to avoid so much rightward indent.

Centril commented on src/librustc/infer/error_reporting/nice_region_error/trait_impl_difference.rs:
```diff
@@ -0,0 +1,53 @@
+                                let sp = var_origin.span();
+                                let mut err = self.tcx().sess.struct_span_err(
+                                    sp,
+                                    "`impl` item doesn't match `trait` item"
```
```suggestion
                                    "`impl` item signature doesn't match `trait` item signature"
```

estebank committed:
review comments

estebank commented:
r? @nikomatsakis

estebank commented on src/librustc/infer/error_reporting/nice_region_error/trait_impl_difference.rs:
```diff
@@ -0,0 +1,56 @@
+                            ValuePairs::Types(sup_expected_found),
+                        ) = (&sub_trace.values, &sup_trace.values) {
+                            if sup_expected_found == sub_expected_found {
+                                let sp = var_origin.span();
```
Is there a way of getting the original trait item's span here? That way we could point at both to make the discrepancy clearer.

nikomatsakis replied:
Hmm. I think that these variables are created in/around this code here:

https://github.com/rust-lang/rust/blob/8c7b921feb5b100b1825ad54d46a8cca575b35b3/src/librustc_typeck/check/compare_method.rs#L234-L238

If that's correct, one option might be to pack more information into the "variable origin". However, that code is *also* using an obligation cause that contains all the information we need:

https://github.com/rust-lang/rust/blob/8c7b921feb5b100b1825ad54d46a8cca575b35b3/src/librustc_typeck/check/compare_method.rs#L93-L101

This is threaded to "subtype" code here:

https://github.com/rust-lang/rust/blob/8c7b921feb5b100b1825ad54d46a8cca575b35b3/src/librustc_typeck/check/compare_method.rs#L290-L294

However, that seems to get lost when converting to a subregion obligation. There is some code tries to preserve it

https://github.com/rust-lang/rust/blob/8c7b921feb5b100b1825ad54d46a8cca575b35b3/src/librustc/infer/mod.rs#L1743-L1765

so I'm not 100% sure what's going on. Would have to dump some logs.

estebank replied:
Should we merge the work done so far and get back to it with more time to add the extra information? The original tickets would remain open.

estebank replied:
The reason is because the failing obligation is of code `ReferenceOutlivesReferent`, not `CompareImplMethodObligation`. We really need to add a new variant to chain obligations together and recover the original problem.

estebank replied:
Scratch that, figured it out.

estebank committed:
Point at the trait item and tweak wording

rust-highfive commented:
Your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=10829&view=logs&jobId=4f922444-fdfe-5dcf-b824-02f86439ef14) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/10829/logs/49), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/10829/logs/49)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-10-15T17:37:18.3630623Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-10-15T17:37:18.3641998Z ##[command]git config gc.auto 0
2019-10-15T17:37:18.3644631Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-10-15T17:37:18.3647204Z ##[command]git config --get-all http.proxy
2019-10-15T17:37:18.3649592Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/65068/merge:refs/remotes/pull/65068/merge
---
2019-10-15T17:39:41.8127766Z Attempting with retry: docker build --rm -t rust-ci -f /home/vsts/work/1/s/src/ci/docker/mingw-check/Dockerfile /home/vsts/work/1/s/src/ci/docker
2019-10-15T17:39:42.1485130Z Sending build context to Docker daemon  526.3kB
2019-10-15T17:39:42.1487527Z 
2019-10-15T17:39:42.1495843Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:42.9816067Z received unexpected HTTP status: 503 Service Unavailable
2019-10-15T17:39:44.0766020Z Sending build context to Docker daemon  526.3kB
2019-10-15T17:39:44.0766992Z 
2019-10-15T17:39:44.0957310Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:44.0957310Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:44.6460506Z received unexpected HTTP status: 503 Service Unavailable
2019-10-15T17:39:46.7446765Z Sending build context to Docker daemon  526.3kB
2019-10-15T17:39:46.7447636Z 
2019-10-15T17:39:46.7635619Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:46.7635619Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:47.3169493Z received unexpected HTTP status: 503 Service Unavailable
2019-10-15T17:39:50.4104493Z Sending build context to Docker daemon  526.3kB
2019-10-15T17:39:50.4104821Z 
2019-10-15T17:39:50.4319267Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:50.4319267Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:50.9751883Z received unexpected HTTP status: 503 Service Unavailable
2019-10-15T17:39:55.0844681Z Sending build context to Docker daemon  526.3kB
2019-10-15T17:39:55.0844829Z 
2019-10-15T17:39:55.1078388Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:55.1078388Z Step 1/6 : FROM ubuntu:16.04
2019-10-15T17:39:55.7365414Z received unexpected HTTP status: 503 Service Unavailable
2019-10-15T17:39:55.7378172Z The command has failed after 5 attempts.
2019-10-15T17:39:55.7526376Z ##[error]Bash exited with code '1'.
2019-10-15T17:39:55.7561160Z ##[section]Starting: Checkout
2019-10-15T17:39:55.7562879Z ==============================================================================
2019-10-15T17:39:55.7562931Z Task         : Get sources
2019-10-15T17:39:55.7563606Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

estebank commented on src/test/ui/reject-specialized-drops-8142.stderr:
```diff
@@ -89,27 +89,16 @@ note: Use same sequence of generic type and region parameters that is on the str
-           expected W<'l1, 'l2>
-              found W<'_, '_>
+   = note: expected `W<'l1, 'l2>`
+              found `W<'_, '_>`
```
This is coming from a `MiscObligation`, which is why it doesn't have a span pointing to the original struct.

estebank committed:
Make error apply only to impl/trait mismatch

estebank committed:
Fix NLL test

JohnCSimon commented:
Ping from triage: 

@nikomatsakis  Can you please review this PR? Thanks
CC: @estebank

nikomatsakis commented:
@bors r+

nice patch, @estebank

bors commented:
:pushpin: Commit 3aba2d060a0184d38c261bee218c973457135aba has been approved by `nikomatsakis`

<!-- @bors r=nikomatsakis 3aba2d060a0184d38c261bee218c973457135aba -->
<!-- homu: {"type":"Approved","sha":"3aba2d060a0184d38c261bee218c973457135aba","approver":"nikomatsakis"} -->

bors commented:
:hourglass: Testing commit 3aba2d060a0184d38c261bee218c973457135aba with merge e097c19d6d8d9564b54923c2903baad9a990bbb2...
<!-- homu: {"type":"BuildStarted","head_sha":"3aba2d060a0184d38c261bee218c973457135aba","merge_sha":"e097c19d6d8d9564b54923c2903baad9a990bbb2"} -->

rust-highfive commented:
The job `dist-i586-gnu-i586-i686-musl` of your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=12196&view=logs&jobId=e8f9fbbd-638d-5b4a-16e9-2519b7fb3143) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/12196/logs/1319), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/12196/logs/1319)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-10-29T01:52:35.9900532Z ---- [ui] ui/in-band-lifetimes/mismatched_trait_impl-2.rs stdout ----
2019-10-29T01:52:35.9900701Z diff of stderr:
2019-10-29T01:52:35.9900751Z 
2019-10-29T01:52:35.9901137Z 3    |
2019-10-29T01:52:35.9901536Z 4 LL |     fn deref(&self) -> &dyn Trait {
2019-10-29T01:52:35.9902274Z 5    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ found fn(&Struct) -> &dyn Trait
2019-10-29T01:52:35.9903111Z -    | 
2019-10-29T01:52:35.9903523Z -   ::: $SRC_DIR/libcore/ops/deref.rs:LL:COL
2019-10-29T01:52:35.9903904Z -    |
2019-10-29T01:52:35.9904288Z - LL |     fn deref(&self) -> &Self::Target;
2019-10-29T01:52:35.9904759Z -    |     --------------------------------- expected fn(&Struct) -> &(dyn Trait + 'static)
2019-10-29T01:52:35.9904963Z 11    |
2019-10-29T01:52:35.9905465Z 12    = note: expected `fn(&Struct) -> &(dyn Trait + 'static)`
2019-10-29T01:52:35.9905708Z 13               found `fn(&Struct) -> &dyn Trait`
2019-10-29T01:52:35.9905924Z 
2019-10-29T01:52:35.9906010Z The actual stderr differed from the expected stderr.
2019-10-29T01:52:35.9906010Z The actual stderr differed from the expected stderr.
2019-10-29T01:52:35.9906395Z Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/in-band-lifetimes/mismatched_trait_impl-2/mismatched_trait_impl-2.stderr
2019-10-29T01:52:35.9906847Z To update references, rerun the tests and pass the `--bless` flag
2019-10-29T01:52:35.9907336Z To only update this specific test, also pass `--test-args in-band-lifetimes/mismatched_trait_impl-2.rs`
2019-10-29T01:52:35.9907622Z error: 1 errors occurred comparing output.
2019-10-29T01:52:35.9907681Z status: exit code: 1
2019-10-29T01:52:35.9907681Z status: exit code: 1
2019-10-29T01:52:35.9908612Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/in-band-lifetimes/mismatched_trait_impl-2.rs" "-Zthreads=1" "--target=i586-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/in-band-lifetimes/mismatched_trait_impl-2" "-Crpath" "-O" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/i586-unknown-linux-gnu/native/rust-test-helpers" "-Clinker=cc" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/in-band-lifetimes/mismatched_trait_impl-2/auxiliary" "-A" "unused"
2019-10-29T01:52:35.9910726Z ------------------------------------------
2019-10-29T01:52:35.9912075Z 
2019-10-29T01:52:35.9912980Z ------------------------------------------
2019-10-29T01:52:35.9913298Z stderr:
2019-10-29T01:52:35.9913298Z stderr:
2019-10-29T01:52:35.9913719Z ------------------------------------------
2019-10-29T01:52:35.9914205Z error: `impl` item signature doesn't match `trait` item signature
2019-10-29T01:52:35.9915766Z   --> /checkout/src/test/ui/in-band-lifetimes/mismatched_trait_impl-2.rs:8:5
2019-10-29T01:52:35.9915888Z    |
2019-10-29T01:52:35.9916160Z LL |     fn deref(&self) -> &dyn Trait {
2019-10-29T01:52:35.9916409Z    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ found fn(&Struct) -> &dyn Trait
2019-10-29T01:52:35.9916641Z    |
2019-10-29T01:52:35.9916919Z    = note: expected `fn(&Struct) -> &(dyn Trait + 'static)`
2019-10-29T01:52:35.9917129Z               found `fn(&Struct) -> &dyn Trait`
2019-10-29T01:52:35.9917241Z error: aborting due to previous error
2019-10-29T01:52:35.9917296Z 
2019-10-29T01:52:35.9917323Z 
2019-10-29T01:52:35.9917526Z ------------------------------------------
---
2019-10-29T01:52:35.9945299Z thread 'main' panicked at 'Some tests failed', src/tools/compiletest/src/main.rs:537:22
2019-10-29T01:52:35.9945462Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-10-29T01:52:35.9966217Z 
2019-10-29T01:52:35.9966498Z 
2019-10-29T01:52:35.9968601Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/i586-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-i586-unknown-linux-gnu" "--mode" "ui" "--target" "i586-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/checkout/obj/build/x86_64-unknown-linux-gnu/llvm/build/bin/FileCheck" "--linker" "cc" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/i586-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--llvm-version" "9.0.0-rust-1.40.0-nightly\n" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-10-29T01:52:35.9969947Z 
2019-10-29T01:52:35.9970008Z 
2019-10-29T01:52:35.9973420Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test --target i586-unknown-linux-gnu,i686-unknown-linux-musl
2019-10-29T01:52:35.9973549Z Build completed unsuccessfully in 1:11:21
2019-10-29T01:52:35.9973549Z Build completed unsuccessfully in 1:11:21
2019-10-29T01:52:36.0030163Z == clock drift check ==
2019-10-29T01:52:36.0045542Z   local time: Tue Oct 29 01:52:36 UTC 2019
2019-10-29T01:52:36.2944608Z   network time: Tue, 29 Oct 2019 01:52:36 GMT
2019-10-29T01:52:36.2945273Z == end clock drift check ==
2019-10-29T01:52:37.5514166Z 
2019-10-29T01:52:37.5610785Z ##[error]Bash exited with code '1'.
2019-10-29T01:52:37.5681214Z ##[section]Starting: Upload CPU usage statistics
2019-10-29T01:52:37.5708195Z ==============================================================================
2019-10-29T01:52:37.5708285Z Task         : Bash
2019-10-29T01:52:37.5708354Z Description  : Run a Bash script on macOS, Linux, or Windows
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

bors commented:
:broken_heart: Test failed - [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12196)
<!-- homu: {"type":"BuildFailed","builder_url":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12196","builder_name":"checks-azure"} -->

estebank committed:
Silence crate external span error in x86 platforms

This causes issues in at least `dist-i586-gnu-i586-i686-musl`,
possibly others.

estebank commented:
@bors r=nikomatsakis

Made that test ignore x86 targets to get around different platform output discrepancies when a `std` span isn't found. There's a ticket open for that already.

bors commented:
:pushpin: Commit 213fd1f37f272e352ceb7dcdaddcd90b1da54d61 has been approved by `nikomatsakis`

<!-- @bors r=nikomatsakis 213fd1f37f272e352ceb7dcdaddcd90b1da54d61 -->
<!-- homu: {"type":"Approved","sha":"213fd1f37f272e352ceb7dcdaddcd90b1da54d61","approver":"nikomatsakis"} -->

bors commented:
:hourglass: Testing commit 213fd1f37f272e352ceb7dcdaddcd90b1da54d61 with merge 0b7e28a1610924a27471ffdb59a2885709b3b415...
<!-- homu: {"type":"BuildStarted","head_sha":"213fd1f37f272e352ceb7dcdaddcd90b1da54d61","merge_sha":"0b7e28a1610924a27471ffdb59a2885709b3b415"} -->

bors commented:
:sunny: Test successful - [checks-azure](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12295)
Approved by: nikomatsakis
Pushing 0b7e28a1610924a27471ffdb59a2885709b3b415 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"nikomatsakis","base_ref":"master","builders":{"checks-azure":"https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_build/results?buildId=12295"},"merge_sha":"0b7e28a1610924a27471ffdb59a2885709b3b415"} -->
